### PR TITLE
[RLlib] Fix Custom Multi Agent example with learner API

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?
This fixes examples/multi_agent_custom_policy_w_rlm_torch/tf2.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -2930,6 +2930,31 @@ def get_marl_module_spec(
                         "is passed in nor in the default module spec used in "
                         "the algorithm."
                     )
+            if module_spec.catalog_class is None:
```
Comment (ArturNiederfahrenhorst) [inline rllib/algorithms/algorithm_config.py:2933]: Further up is a typo in L2905:

# PolicySpecs and the **defaul** RLModuleSpec.

**Review by ArturNiederfahrenhorst — COMMENTED** [review]
Thanks for these changes. Everything looks clean to me. The only nit is for adjacent code so I don't request a change.

You'll only need to fix a couple of tests (for example test_catalog.py and tests_algorithm.py) that don't meet the requirements to the specs anymore now.

**Comment by kouroshHakha** [comment]
> Thanks for these changes. Everything looks clean to me. The only nit is for adjacent code so I don't request a change.
> 
> You'll only need to fix a couple of tests (for example test_catalog.py and tests_algorithm.py) that don't meet the requirements to the specs anymore now.

@ArturNiederfahrenhorst This happened because of the functionality merge conflict that your code and my code introduces when they got on master. The master is actually red for both of them now. This should clear that up.

**Review by gjoliver — APPROVED** [review]

File: rllib/algorithms/algorithm_config.py [hunk 2]
```diff
@@ -2930,6 +2930,31 @@ def get_marl_module_spec(
...
+                if isinstance(default_spec, SingleAgentRLModuleSpec):
+                    module_spec.catalog_class = default_spec.catalog_class
+                elif isinstance(default_spec.module_specs, SingleAgentRLModuleSpec):
+                    catalog_class = default_spec.module_specs.catalog_class
+                    # This should be already checked in validate() but we check it
+                    # again here just in case
+                    if catalog_class is None:
+                        raise ValueError(
+                            "The default rl_module spec cannot have an empty "
```
Comment (avnishn) [inline rllib/algorithms/algorithm_config.py]: the default marl module spec *. Right? I assume we only enter this branch if the `default_spec` is a `MultiAgentRLModuleSpec`

**Review by avnishn — COMMENTED** [review]
have a q

File: rllib/algorithms/algorithm_config.py (hunk 2 above)
Comment (kouroshHakha) [reply rllib/algorithms/algorithm_config.py]: Not really. This is the RLModule spec with in the default MARLSpec.

**Pull request merged**